### ISSUE TITLE
 transpose([2, 0, 1])

### ISSUE THREAD

**Issue by lx123-j** [Hwihuni/BlindHarmony#1]
Hello! I would like to ask why the data is being transformed using transpose([2, 0, 1]). If my data is initially in the shape of (1, 256, 256), do I still need to perform this transformation?

**Comment by Hwihuni** [comment]
Hello! Most MRI data have a structure where the last axis corresponds to the slice direction, so I simply reshape it to process the image data with a slice level. In your case, it is already in a slice level (or you want another modality), so you don't have to transform your data.

**Comment by lx123-j** [comment]
Thank you for your response. After running the tests, I encountered a new issue where the results exhibit significant artifacts. Is this considered normal? Additionally, I would like to establish a one-to-one correspondence between the input images and their respective test results. It seems that we did not specify the exact names of the input images during input. How can I associate the co-registered images with the source domain images?

**Comment by Hwihuni** [comment]
For your first question, I think you mentioned 'exploding inverse' artifacts. It is not a normal situation, but quite a common artifact in normalizing flow (see https://proceedings.mlr.press/v130/behrmann21a.html).

For your second question, I set the initial image (which you mentioned as 'input image') as an averaged image (in a slice level) of training data. You can choose any other images such as source domain image itself.

**Comment by lx123-j** [comment]
Thank you very much for your response. However, I may not have expressed myself clearly. In fact, what I want to ask is how we can ensure that for each input image from our source domain, we get the corresponding co-registered output image. Since the input pckl_sc directly extracts pklv4 files which do not contain file names, when there is a set of source domain images, how can we determine which co-registered output image corresponds to which input image? Is it because the order of the output images matches the order of the input images? Additionally, I encountered an issue with assert(discriminant>=0).all(). How should I address this problem during training or testing? Could it be related to insufficient preprocessing of my data?

**Comment by Hwihuni** [comment]
Sorry for replying late! In my code, I assumed that the order of the output images matches the that of input. You can rewrite my code to save the image with the designated file name, For the issue with assert(discriminant>=0).all(), I recommend you to use smaller learning rate.
